crates/core/src/client_events/mod.rs

```diff
@@ -356,10 +360,97 @@ enum Error {
 }
 
 #[inline]
+async fn handle_proximity_cache_info_query(
+    proximity_cache: &Arc<crate::node::proximity_cache::ProximityCacheManager>,
+) -> freenet_stdlib::client_api::ProximityCacheInfo {
+    let (my_cache_contract_ids, neighbor_cache_data) =
+        proximity_cache.get_introspection_data().await;
+    let stats = proximity_cache.get_stats().await;
+
+    let my_cache = my_cache_contract_ids
+        .into_iter()
+        .map(|contract_id| {
+            use freenet_stdlib::prelude::ContractKey;
+            let contract_key = ContractKey::from(contract_id);
+            // Use first 4 bytes for backwards-compatible cache_hash field
+            let bytes = contract_id.as_bytes();
+            let cache_hash = u32::from_le_bytes([
+                bytes.first().copied().unwrap_or(0),
+                bytes.get(1).copied().unwrap_or(0),
+                bytes.get(2).copied().unwrap_or(0),
+                bytes.get(3).copied().unwrap_or(0),
+            ]);
+
+            freenet_stdlib::client_api::ContractCacheEntry {
+                contract_key: contract_key.to_string(),
+                cache_hash,
+                cached_since: std::time::SystemTime::now()
```

Comment:
`cached_since` and `neighbor.last_update` are being set to `SystemTime::now()` at query time, so every CLI call reports fresh timestamps even if nothing changed. That makes the introspection data look useful but it is effectively random. Could we either carry real timestamps from the cache manager or drop these fields for now so we do not mislead operators?